### PR TITLE
Add Trino 463 release notes

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Assemble the release notes for the upcoming Trino 463 release.

## Additional context and related issues

See https://github.com/trinodb/trino/pulls?q=is%3Apr+is%3Aclosed+milestone%3A463

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

## Verification for each pull request

Format: PR/issue number, ✅ / ❌ rn ✅ / ❌ docs
✅ rn - release note added and verified, or assessed to be not necessary, set to ❌ rn before completion
✅ docs - need for docs assessed and merged, or assessed to be not necessary, set to ❌ docs before completion

## 16 Oct 2024

* #23811 ✅ rn ✅ docs
* #23809 ✅ rn ✅ docs

## 17 Oct 2024

* #23564 ✅ rn ✅ docs, see #23821
* #23741 ✅ rn ✅ docs
* #23816 ✅ rn ✅ docs
* #23459 ✅ rn ✅ docs
* #23801 ✅ rn ✅ docs
* #23821 ✅ rn ✅ docs
* #23744 ✅ rn ✅ docs

## 18 Oct 2024

* #23825 ✅ rn ✅ docs
* #23806 ✅ rn ✅ docs
* #23393 ✅ rn ✅ docs
* #23777 ✅ rn ✅ docs
* #21000 ✅ rn ✅ docs
* #15337 ✅ rn ✅ docs

## 19 Oct 2024

* #23840 ✅ rn ✅ docs
* #23814 ✅ rn ✅ docs

## 20 Oct 2024

* #23453 ✅ rn ✅ docs
* #23830 ✅ rn ✅ docs
* #23834 ✅ rn ✅ docs
* #23841 ✅ rn ✅ docs

## 21 Oct 2024

* #23848 ✅ rn ✅ docs
* #23856 ✅ rn ✅ docs
* #23831 ✅ rn ✅ docs

## 22 Oct 2024

* #23850 ✅ rn ✅ docs
* #23861 ✅ rn ✅ docs
* #23827 ✅ rn ✅ docs
* #23858 ✅ rn ✅ docs
* #23863 ✅ rn ✅ docs
* #23865 ✅ rn ✅ docs
* #23723 ✅ rn ✅ docs
* #23868 ✅ rn ✅ docs
* #23857 ✅ rn ✅ docs

## 23 Oct 2024

* #23844 ✅ rn ✅ docs
* #23873 ✅ rn ✅ docs
* #23872 ✅ rn ✅ docs
* #23874 ✅ rn ✅ docs
* #23797 ✅ rn ✅ docs
* #23684 ✅ rn ✅ docs

### CODE REVIEW

**Comment by wendigo** [comment]
@mosabua 463 ;)

File: docs/src/main/sphinx/release/release-463.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+# Release 463 (23 Oct 2024)
+
+## General
+
+* Add {func}`time_zone` functions to extract the IANA timezone string from
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-463.md:5]: fyi @wendigo

File: docs/src/main/sphinx/release/release-463.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+## BigQuery connector
+
+## Blackhole connector
+
+## Cassandra connector
+
+## ClickHouse connector
+
+* Improve performance for queries with `IS NULL` expressions. ({issue}`23459`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-463.md:30]: fyi @ssheikin and @Praveen2112

**Comment by mosabua** [comment]
> @mosabua 463 ;)

I know .. haha

File: docs/src/main/sphinx/release/release-463.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+## Delta Lake connector
+
+* Add support for writing change data feed when [deletion vector](https://docs.delta.io/latest/delta-deletion-vectors.html) 
+  is enabled. ({issue}`23620`)
+
+## Iceberg connector
+
+* Add support for nested namespaces with the REST catalog. ({issue}`22916`)
+* Add support for configuring the maximum number of rows per row-group in the
+  ORC writer with the `orc_writer_max_row_group_rows` catalog session property. ({issue}`23722`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-463.md:25]: Added the name of the property like in earlier release notes @raunaqmorarka  ... please confirm that it is a catalog session property.

Also we dont want docs for any of them?

And they are Iceberg only?

File: docs/src/main/sphinx/release/release-463.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+# Release 463 (23 Oct 2024)
+
+## General
+
+* Add {func}`timezone` functions to extract the timezone identifier from a
+  timestamp or time. ({issue}`20893`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-463.md:6]: Maybe say "from a `timestamp(p) with time zone` or `time(p) with time zone`.

File: docs/src/main/sphinx/release/release-463.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+* Add table functions to the `SHOW FUNCTIONS` output. ({issue}`12550`)
```
Comment (martint) [inline docs/src/main/sphinx/release/release-463.md:7]: "List table functions with `SHOW FUNCTIONS`"

File: docs/src/main/sphinx/release/release-463.md (hunk 4 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-463.md:6]: I was contemplating .. lets add that detail.

File: docs/src/main/sphinx/release/release-463.md (hunk 5 above)
Comment (mosabua) [reply docs/src/main/sphinx/release/release-463.md:7]: yeah .. but thats kinda misleading .. its also listing table functions..

Comment (mosabua) [reply docs/src/main/sphinx/release/release-463.md:7]: Include table functions with SHOW FUNCTIONS output. ??

Comment (martint) [reply docs/src/main/sphinx/release/release-463.md:7]: Well, it doesn’t say “list ONLY table functions…”, it just implies that the command can now list table functions. But fair enough. The latter phrasing you suggested works, too.

File: docs/src/main/sphinx/release/release-463.md (hunk 3 above)
Comment (raunaqmorarka) [reply docs/src/main/sphinx/release/release-463.md:25]: it is a catalog session property in iceberg.
i would skip documentation for now, it's very technical detail

File: docs/src/main/sphinx/release/release-463.md [hunk 6]
```diff
@@ -0,0 +1,35 @@
+# Release 463 (23 Oct 2024)
+
+## General
+
+* Enable HTTP/2 for internal communication by default. ({issue}`21793`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-463.md:5]: @wendigo do we also want to add that HTTP/2 support for clients is there now? Also .. any config we need to add to the release notes as fallback or do we want to link somewhere?

Comment (martint) [reply docs/src/main/sphinx/release/release-463.md:5]: Maybe just say "the previous behavior can be restored by setting xxx to yyy".

Comment (mosabua) [reply docs/src/main/sphinx/release/release-463.md:5]: Ok .. I will also send a docs PR but ideally we get #23832 and #23833 merged as well so I can cleanly add a new page for internal communication properties.

**Review by martint — APPROVED** [review]

**Pull request merged**